models/payout.go

```diff
@@ -56,6 +56,19 @@ func AllocatePayouts() error {
 			return err
 		}
 
+		for _, l := range ledgerEntries {
+			if l.authorId == user.ID {
+				// deposit ledgers
+				return createLedgerEntry(user.ID, l.contributorId, user.Cash, l.ledgerType, l.description)
```

Comment:
We do this because on line 24, we don't have access to contributor data for each upvote the audience votes to the artist. Since we're storing contributor field in the ledger table I chose this way where we create ledger entries soon after we execute the stripe transfer. Hope it makes sense. Do let me know if you think there's a better approach.

Reply:
Got it - that's correct. I think originally I was trying to do it this way to create less writes on the DB, but in the end I think it just adds too much complexity since we'll be doing the same amount of writes with the ledger. I'd love your thoughts on this approach instead:

Instead of calculatePayouts returning both the ledger and the aggregate sums each user is getting paid, what if we only return the ledger? Then in AllocatePayouts we iterate through each ledger entry and create a transaction that both:
1. updates the user's cash in the user table
2. adds a ledger entry that mirrors this

It might be a more simple flow here.

Reply:
There might be a loophole here. If we return only the ledger entries as an array, each entry will have a fraction of the user's cash contributed by the audience and not the whole amount. In that case we again need to add up the fractions of cash(in a loop again) upvoted by the audience and finally update the user's cash in the user's table. I think there's no skipping of that functionality. We would be calculating the payout again in `AllocatePayouts()` then. Lemme know if this makes sense